### DISCUSSION TITLE
equivalent of using formikcontext in react-hook-form?

### DISCUSSION THREAD

**Discussion by kamigerami** [react-hook-form/react-hook-form#8750] [Q&A]
Hey,

So I have a quick question. I have this datepicker / calendar component that is an external lib and I have no control over. I can't pass any {...register} to it nor {..fields} so what would the equivalent solution be in react-hook-form that is here in this codepen 
https://codesandbox.io/s/serene-fog-oob88-oob88?fontsize=14&hidenavigation=1&theme=dark&file=/src/index.js:413-496

Thnx

**Comment by Moshyfawn** [comment]
[`FormProvider` + `useFormContext`](https://react-hook-form.com/api/useformcontext) for uncontrolled components _(using `register`)_ or `useController` with either `FromProvider` at the form's root or simply pass `control` into it manually

**Reply by kamigerami** [reply]
thank you for the suggestion, tried it out - It seems like setting the field.onChange to the onChange of the component breaks it (can't select range of dates using the calendarPicker anymore)


```

<FormProvider {...methods} >      
<form onSubmit={handleSubmit(onSubmit)}>
              <Calendar
                value={selectedDayRange}
                onChange={() => field.onChange(setSelectedDayRange)}
              />
       </form>
    </FormProvider>             
 ```
 Also it does not accept any props such as name etc ... so I can't really follow the guides on the react-hook-form page. Any other suggestions on how to solve this? thnx

**Reply by kamigerami** [reply]
I think I might abandon that particular Datepicker and solve my problems that way :) react-hook-form is great thnx a lot!
